File: Resueltos_por_competidor/Mariana_Pineda/SoldierAndBananas.cpp
#include <iostream>

using namespace std;
/*
	 * Autor: mariana 
	 * Problema : Soldier and Bananas 
	 * Juez online: codeforces
	 * Veredicto: Accepted 
	 * URL: https://codeforces.com/contest/546/problem/A
	 */
int main() {
    int n;
    int k;
    int w;
    int total=0;
    cin>>k;
    cin>>n;
    cin>>w;
    total = k * ((w)*(w+1)/2);
    if(total-n<0){
        cout<<0;
    }else{
    cout<<total-n;
    }
    }

File: Resueltos_por_competidor/Santiago_Ortiz/SoldierAndBananas.java
import java.util.Scanner;

/*
 * Autor: Darkovski7   
 * Problema: Soldier and Bananas
 * Juez Online: Codeforces
 * Veredicto: Accepted 
 * Url: https://codeforces.com/problemset/problem/546/A
 */

public class SoldierAndBananas {

	public static void main(String[] args) {

		Scanner sc = new Scanner(System.in);
		
		int k = sc.nextInt();
		int n = sc.nextInt();
		int w = sc.nextInt();
		int total = 0;
		int resultado = 0;
		
		for (int i = 1; i < w+1; i++) {
			total += k*i;
			resultado = total - n;
		}
		if(total<n) {
			System.out.println(0);
		} else {
		System.out.println(resultado);
	}
	}
}